### ISSUE TITLE
session.list vs theoretically-same SQLQuery differences

### ISSUE THREAD

**Issue by Sylvyrfysh** [vsch/kotlin-jdbc#10]
This appears to be related to how the model.list with the where clause is created. The SQLQuery that gets created internally has the database quoting appended to it in a list, rather than just the params map getting passed in. I I have time, I'll submit a fix. Her is some code that would create the issue

```Kotlin
                val query = sqlQuery("""
                    select * from ${model.tableName}
                    where date >= :startDate AND date <= :endDate;
                """.trimIndent(), mapOf("startDate" to startDate, "endDate" to endDate))

            val item = model.list(session, "where date >= :startDate AND date <= :endDate", mapOf("startDate" to startDate, "endDate" to endDate))

            println(item)

            val dataset = session.list(query, model.toData).sorted()

            println(dataset)
```

dataset will have items with valid data, itemwill be empty.

**Comment by Sylvyrfysh** [comment]
Partially fixed in [my branch here](https://github.com/Sylvyrfysh/kotlin-jdbc/tree/sqlquery-quote-fixes)

**Comment by Sylvyrfysh** [comment]
Fixed in #11

**Comment by vsch** [comment]
@Sylvyrfysh, thank you for the fixes. I have not looked at the master branch for a looong time. So was a bit confused when looking at the code. Then remembered that the master branch is still on 0.4.x.

I have been using the dev-0.5.0 branch with the new models which are much more convenient: https://github.com/vsch/kotlin-jdbc/blob/dev-0.5.0/README.md. 

I also changed the scripts in that branch to look for `model-config.json` up the directory tree from selected location until a directory `.idea` is found, meaning most likely top of project reached.

If found, this file provides configuration for generating models with correct package information and location in the project. It is no longer necessary to select specific directories for specific tables when generating the model, then to edit the model for the correct package.

I define this file at the project root level and make sure all models which go to non-standard directories are added. To generate a model I only need to choose the tables and any sub-directory in the project to get the model generated in the right location.

I also fixed the scripts to properly get nullability from the database so manual editing to the generated model is minimized.

I was working on a project where the models are all over the place and I got tired of having to tip-toe around model generation and then edit it manually each time.

ℹ️ **I will add the indent string setting to that file so that the script could be used without modification. The config file will provide the indentation for each project with default fall back to four spaces.**

I think it is time to move version 0.4 from master to 0.4.0 branch and move dev-0.5.0 to master. I do not see going back to the old models and there are too many fixes and changes to bother with keeping both branches up to date.

**Comment by Sylvyrfysh** [comment]
Sounds good! Will you be able to move these patched to 0.5.0, or will you need patches for that as well? I'll move to 0.5.0 when I have the time, more important that I get this (small) project done and then I can move it to the new system.

**Comment by vsch** [comment]
@Sylvyrfysh, already patched in latest release. 

What confused me originally was that I already changed the indent to a variable in the script in dev-0.5.0 branch.

I added the overload to helpers but the model list method has a new implementation that does not allow for where clause with varargs Pair params, only map.

The scrips in latest repo version override all script hardcoded values from the `model-config.json` file if they are defined there. Script configuration can now be provided through this file, eliminating the need to change the script. 

Changing the script is problematic because it is shared across projects. It is only possible to have project specific model scripting when using `model-config.json` overrides, not by changing the script manually.
